docs/src/conventions_internal.md

```diff
@@ -0,0 +1,28 @@
+# Conventions used internally
```

Comment:
I really like this write-up, thank you very much! This should make it also much clearer for new users who are just starting with Trixi...

However, I am wondering if we really want this separation into user-facing and internal conventions - where exactly do you draw the line here? Aren't the "variable names" also something that is mainly internally relevant? Also, cells vs. elements vs. nodes is arguably just an internal convention to remain sane when dealing with all three concepts simultaneously in a single code section, but nothing the "user" really needs to worry about (but here you could also argue otherwise, I admit).

I'd probably just tack the new section `Array types and wrapping` to the end of the current `conventions.md` and maybe add a `!!! note` that this is mainly relevant for internal code development.

Reply:
That's a good point but I get the impression that this discussion is drifting far away from the original intend of this PR. I will do what you suggested but also try to re-arrange some parts of the docs, trying to improve the experience for people new to Trixi. Right now, "Conventions" is basically the second part of the docs, which is probably way too early for people without any knowledge of Trixi.

Reply:
Good point. Just leave it as it is then and we'll pick this up in a different issue/PR.